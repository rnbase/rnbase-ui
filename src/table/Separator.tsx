import React from 'react';
import { StyleSheet, View, ViewProps } from 'react-native';

import { Theme, Themed, withTheme } from '../theming';

interface SeparatorProps extends ViewProps {}

const Separator: React.FC<Themed<typeof createStyleSheet, SeparatorProps>> = ({
  theme: { styles },
  style,
  ...props
}) => <View {...props} style={[styles.root, style]} />;

const createStyleSheet = ({ colors }: Theme) =>
  StyleSheet.create({
    root: {
      marginLeft: 20,
      height: StyleSheet.hairlineWidth,
      backgroundColor: colors.separator,
    },
  });

export default withTheme(Separator, createStyleSheet, 'Table.Separator');
