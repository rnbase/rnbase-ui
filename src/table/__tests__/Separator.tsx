import React from 'react';

// Note: test renderer must be required after react-native.
import TestRenderer from 'react-test-renderer';

/**
 * Under test
 */
import Separator from '../Separator';

type Props = React.ComponentProps<typeof Separator>;

const createElement = (props: Partial<Props>) => <Separator {...props} />;

const createRenderer = (props: Partial<Props>) => TestRenderer.create(createElement(props));

it('should render normally', () => {
  expect(createRenderer({})).toMatchSnapshot();
});

it('should render as highlighted', () => {
  expect(createRenderer({ highlighted: true })).toMatchSnapshot();
});

it('should render with custom inset', () => {
  expect(createRenderer({ insetLeft: 123 })).toMatchSnapshot();
});

it('should render with custom styles', () => {
  expect(createRenderer({ style: { backgroundColor: 'red' } })).toMatchSnapshot();
});
