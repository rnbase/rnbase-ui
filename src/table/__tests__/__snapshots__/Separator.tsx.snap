// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as highlighted 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 20,
      },
      Object {
        "backgroundColor": "#0003",
        "height": 0.5,
      },
      undefined,
      Object {
        "opacity": 0,
      },
    ]
  }
/>
`;

exports[`should render normally 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 20,
      },
      Object {
        "backgroundColor": "#0003",
        "height": 0.5,
      },
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`should render with custom inset 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 123,
      },
      Object {
        "backgroundColor": "#0003",
        "height": 0.5,
      },
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`should render with custom styles 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 20,
      },
      Object {
        "backgroundColor": "#0003",
        "height": 0.5,
      },
      Object {
        "backgroundColor": "red",
      },
      undefined,
    ]
  }
/>
`;
